docs/TEMPORARY_DIRECTORIES.md

```diff
@@ -0,0 +1,201 @@
+   intervals, use `utimensat()` or a related call to update the access time
+   ("atime") of all files that shall be kept around. Since the ageing algorithm
+   looks at the access time of files when deciding whether to delete them, it's
+   sufficient to update their access times in sufficiently frequent intervals to
+   ensure the files are not deleted. Since most applications (and tools such as
+   `ls`) primarily care for the modification time (rather than the access time)
+   using the access time for this purpose should be acceptable.
+
+5. Set the "sticky" bit on regular files. The ageing logic skips deletion of
+   all regular files that have the sticky bit (`chmod +t`) set. This is
+   honoured for regular files only however, and has no effect on directories as
+   the sticky bit has a different meaning for them.
+
+6. Don't use `/tmp/` or `/var/tmp/`, but use you own sub-directory under `/run/`
+   or `$XDG_RUNTIME_DIRECTORY` (the former if privileged, the latter if
+   unprivileged), or `/var/lib/` and `~/.config/` (similar, but with
+   persistency and suitable for larger data). The two temporary directories
+   `/tmp/` and `/var/tmp/` come with the implicit clean-up semantics described
+   above. When this is not desired, it's possible to create private per-package
+   runtime or state directories, and place all temporary files there. However,
+   do note that this means opting out of any kind of automatic clean-up, and it
+   is hence particularly essential that the program cleans up generated files
+   in these directories when they are no longer needed, in particular when the
+   program dies abnormally.
+
+5. Exclude your temporary files from clean-ups via a `tmpfiles.d/` drop-in. The
+   `x`/`X` line types may be used to exclude files matching the specified
+   globbing patterns from the ageing logic. If this is used, automatic clean-up
+   is not done for matching files and directory, and much like with the
+   previous option it's hence essential that the program generating these
+   temporary files carefully removes the temporary files it creates again, and
+   in particular so if it dies abnormally.
+
+🥇 The semantics of options 2 (in case you only deal with temporary files, not
+directories) and 3 (in case you deal with both) in the list above are in most
+cases the most preferable. It is thus recommended to stick to these two
+options.
+
+While the ageing logic is very useful as a safety concept to ensure unused
+files and directories are eventually removed a well written program avoids even
+creating files that need such a clean-up. In particular:
+
+1. Use `memfd_create()` or `O_TMPFILE` when creating temporary files. Files
+   created this way never exist linked behind file names in the file system. This
+   means they never have to be deleted, and are automatically released by the
+   kernel when they are fully closed (including when all processes using them
+   exit).
```

Comment:
This repeats what was already said above. I think everything after the first sentence should go.